mesa/extra.py

```diff
@@ -58,13 +59,33 @@ def social_proximity(ego:Person, alter:Person):
     acc += 1 if [x for x in alter.neighbors.get("friendship") if (x in ego.neighbors.get("friendship"))] else 0
     return acc
 
-def at_most(agentset, n, rng_istance):
+
+def at_most(agentset: Union[List[Person], Set[Person]], n: int, rng_istance: numpy.random.default_rng) -> Union[List[Person], Set[Person]]:
+    """
+    Given an @agentset and an integer @n, this function returns the initial @agentset if there are
+    less than @n agents, a subset of those agents of length @n if there are more than @n agents.
+    :param agentset: Union[List[Person], Set[Person]]
+    :param n: int
+    :param rng_istance: numpy.random.default_rng
+    :return: Union[List[Person], Set[Person]]
+    """
     if len(agentset) < n:
         return agentset
     else:
         return list(rng_istance.choice(agentset, n, replace=False))
 
-def weighted_n_of(n, agentset, weight_function, rng_istance):
+def weighted_n_of(n: int, agentset: Union[List[Person], Set[Person]],
+                  weight_function: Callable, rng_istance: numpy.random.default_rng) -> List[Person]:
+    """
+    Given a set or List of agents @agentset, an integer @n, and a lambda function @weight_function.
+    This function performs a weighted extraction, without replacing based on the lambda function.
+    This procedure takes into account negative numbers and weights equal to zero.
+    :param n: int
```

Comment:
I don't understand the need for these :param annotations. The type is already expressed up there. This can't be why the :param is here.. we need to talk about this I guess

Reply:
I mainly used the Sphinx style for docstrings, https://www.sphinx-doc.org/en/master/index.html